cairo/tests/utils/strategies.py

```diff
@@ -69,6 +69,45 @@ def st_from_type(cls):
     ).map(lambda x: cls(**x))
 
 
+def trie_strategy(thing):
+    key_type, value_type = thing.__args__
+
+    # Create a subclass of Trie that preserves the type parameters
```

Comment:
solution I came up with when thinking about how to access the proper types in the tests. Basically store them as class attributes so that get_type() returns Trie[instance.key_type, instance.value_type].

Otherwise we're having issues for Optional[T] types